pytorch_lightning/callbacks/base.py

```diff
@@ -29,6 +29,14 @@ class Callback(abc.ABC):
     Subclass this class and override any of the relevant hooks
     """
 
+    @property
+    def state_identifier(self) -> str:
+        return self.__class__.__name__
```

Comment:
from the linked issue, i'm curious what implementation you're going with for the stateful callbacks that come with pytorch lightning. i'm assuming you'll send this out in a separate PR